interfaces/builtin/dummy.go

```diff
@@ -0,0 +1,99 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/snap"
+)
+
+type testSnapdInterface struct{}
+
+const testSnapdInterfaceSummary = `snapd dummy test interface`
+const testSnapdInterfaceBaseDeclarationSlots = `
+  dummy:
+    allow-installation:
+      slot-snap-type:
+        - app
+    deny-auto-connection: true
+    deny-connection:
+      on-classic: false
+`
+
+func (iface *testSnapdInterface) String() string {
+	return iface.Name()
+}
+
+// Name returns the name of the dummy interface.
+func (iface *testSnapdInterface) Name() string {
+	return "dummy"
+}
+
+func (iface *testSnapdInterface) StaticInfo() interfaces.StaticInfo {
+	return interfaces.StaticInfo{
+		Summary:              testSnapdInterfaceSummary,
+		BaseDeclarationSlots: testSnapdInterfaceBaseDeclarationSlots,
+	}
+}
+
+func (iface *testSnapdInterface) BeforePreparePlug(plug *snap.PlugInfo) error {
+	return nil
+}
+
+func (iface *testSnapdInterface) BeforePrepareSlot(slot *snap.SlotInfo) error {
+	return nil
+}
+
+func (iface *testSnapdInterface) BeforeConnectPlug(plug *interfaces.ConnectedPlug) error {
+	var value string
+	if err := plug.Attr("consumer-attr-3", &value); err != nil {
```

Comment:
Even being a test interface, ideally its attributes would look nice and give a hint of their meaning.

How about renaming both of those attributes to _"before-connect"_ (both the same), and then doing the value replacement like this: 
```
value = fmt.Sprintf("plug-changed(%v)", value)
```
and the equivalent for the slot.

Besides looking nice, this makes sure that we handle matching names correctly (while we don't generally want that in real interfaces, each of these attributes needs to live in their own namespace), and we also make sure that the logic was provided to the right end of it (with "foo-validated" they are both doing the same change).

Reply:
Thanks, that's a good suggestion. done.